artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ActiveMQServerImpl.java

```diff
@@ -294,7 +294,7 @@
 
    private FileStoreMonitor fileStoreMonitor;
 
-   private final Map<String, ServerSession> sessions = new ConcurrentHashMap<>();
+   private final ConcurrentMap<String, ServerSession> sessions = new ConcurrentHashMap<>();
```

Comment:
Can a perf test or some other benefit for this change be provided. This is core parts of the broker where historically concurrent issues introduced by such small changes have had tragic effects

Reply:
@michaelandrepearce I tried to explain the reasoning for the change as part of the Jira issue, have you read that already?

Reply:
So our concerns are this is in core. Its an un-proven optimization, that without any proven hot path benefit brings alot of risk, as small changes like this have previously introduced unexpected issues at first not obvious

Reply:
ATM I am not really sure how I can provide a perf test for the admin UI. The issue is that getSessions() can be very 

 The getSessions() methods are only used by the ActiveMQServerControlImpl and the ConnectionView class, so I would think that this particular change cannot negatively affect the hot path.

ATM I am not really sure how I can provide a perf test for the effect in the webconsole.

The getSessions() methods are only used by the ActiveMQServerControlImpl and the ConnectionView class to list session attributes or to close a session manually, so I can't see how removing the synchronized keyword from these methods can affect the performance of the hot path negatively.

Also if this synchronization really is required than something else must already be broken here since ActiveMQServerImpl.removeSession(), ActiveMQServerImpl.createSession() and ActiveMQServerImpl.getSessionByID() access/modify the sessions map without synchronizing on `this`.

So even without a perf test I would argue something should be changed, either createSession/getSessionByID/removeSession also need to synchronize on `this` or the `synchronized` keyword on `getSessions()` methods is useless.

Reply:
@michaelandrepearce I attached a testcase to the JIRA issue where I try to highlight the impact of this PR. I hope this satisfies your request.

Reply:
So idea is to include test cases etc in PRs so they live in the repo